### PR TITLE
Feature/group chat

### PR DESCRIPTION
**Pull request by yermolenko-d**

### CODE REVIEW

**Comment by Potapy4** [comment]
why do we need RabbitMQ for communication?

**Comment by yermolenko-d** [comment]
> why do we need RabbitMQ for communication?

Because we store messages in db and I decided to implement it like we've implement notifications or build statuses before.

File: frontend/src/app/modules/group/group-chat/group-chat.component.sass [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      max-width: 350px
+      word-wrap: break-word
+
+      &:before
+        right: -12px
+        border-bottom: 20px solid transparent
+        border-left: 20px solid #d84848
+
+
+button
```
Comment (VoBilyk) [inline frontend/src/app/modules/group/group-chat/group-chat.component.sass:91]: Better to style class, but not an element

File: frontend/src/app/modules/group/group-chat/group-chat.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      this.groupId = data.group.id;
+      });
+    this.user = this.auth.getCurrentUser();
+   }
+
+  ngOnInit(): void {
+    this.chat.buildConnection();
+    this.chat.startConnectionAndJoinGroup(this.groupId.toString());
+    this.chat.messageListener(this.message);
+    this.message.subscribe((res) => {
```
Comment (VoBilyk) [inline frontend/src/app/modules/group/group-chat/group-chat.component.ts:45]: What will happen when component is destroyed and created again? Need to unsubscribe if you don't want to get duplications

File: frontend/src/app/modules/group/group-chat/group-chat.component.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  sendMessage(){
+    const msg = new Message();
+    console.log(this.textOfMessage);
+    msg.text = this.textOfMessage;
+    msg.groupId = this.groupId;
+    msg.senderId = this.user.id;
+    this.chat.sendMessage(msg).subscribe((res) => console.log());
+    this.textOfMessage = '';
```
Comment (VoBilyk) [inline frontend/src/app/modules/group/group-chat/group-chat.component.ts:75]: You'd better use an interface and simplify object initialization:
const msg: Message = {
   text: this.textOfMessage,
   groupId = this.groupId,
   senderId = this.user.id
};

**Review by VoBilyk — CHANGES_REQUESTED** [review]

**Comment by VoBilyk** [comment]
Left a few remarks

File: frontend/src/app/modules/group/group-chat/group-chat.component.sass (hunk 1 above)
Comment (yermolenko-d) [reply frontend/src/app/modules/group/group-chat/group-chat.component.sass:91]: ok, but it was only 1 button on a page so I decided that it's ok
But I will change it if you say so

**Review by VoBilyk — APPROVED** [review]

**Pull request merged**